File: android/src/main/java/com/musiclibrary/tracks/GetTracksByArtistQuery.kt
package com.musiclibrary.tracks

import android.content.ContentResolver
import com.musiclibrary.models.PaginatedResult
import com.musiclibrary.models.Track
import com.musiclibrary.models.TrackOptions

object GetTracksByArtistQuery {
  fun getTracksByArtist(
    contentResolver: ContentResolver,
    artistId: String,
    options: TrackOptions,
  ): PaginatedResult<Track> {
    return TrackQuery.getPaginatedTracks(
      contentResolver = contentResolver,
      filter = TrackQueryFilter.Artist(artistId),
      options = options
    )
  }
}

File: ios/tracks/GetTracksByArtistQuery.swift
import Foundation

internal class GetTracksByArtistQuery {
  static func getTracksByArtist(artistId: String, options: TrackOptions) throws -> PaginatedResult<Track> {
    return try TrackQuery.getPaginatedTracks(filter: .artist(artistId), options: options)
  }
}
